docs/design/coreclr/botr/readytorun-format.md

```diff
@@ -428,6 +475,31 @@ The module override index translation algorithm is as follows (**ILAR** = *the n
 
 **TODO**: document attribute presence encoding
 
+**Note**: We already know this table uses assembly-relative token encoding so it has similar
+characteristics like `READYTORUN_SECTION_AVAILABLE_TYPES` or `READYTORUN_SECTION_METHOD_ENTRYPOINTS`.
+No matter what component assembly-relative encoding we end up choosing for these tables, we
+should use the same encoding for ATTRIBUTEPRESENCE.
+
+## READYTORUN_SECTION_ASSEMBLIES (v3.2+)
+
+This section is only present in composite R2R files. It is a straight binary array of the
+entries `READYTORUN_SECTION_ASSEMBLIES_ENTRY` parallel to the indices in the manifest metadata
+AssemblyRef table in the sense that it's a linear table where the row indices correspond to the
+equivalent AssemblyRef indices. Just like in the AssemblyRef ECMA 335 table, the indexing is
+1-based (the first entry in the table corresponds to index 1).
+
+```C++
+struct READYTORUN_SECTION_ASSEMBLIES_ENTRY
```

Comment:
This shouldn't be a hard-coded list of items what are assembly local. We probably need this to be a variable sized structure, described by section entries just like the outer section concept.

Reply:
I think you're right, I've also been thinking about this when resurrecting my doc from last summer. I'll update the document based on your suggestion.

Reply:
The only trick is how to pull this off without introducing an unintentional <code>O(n)</code> lookup when searching for, say, the per-module entrypoint table - but you may argue that <code>n</code> is usually very small (less than ten with a huge headroom).

Reply:
Mmm.. possibly. I rather imagine it won't really be an O(n) loop at lookup time, more of a thing where we capture all of the interesting data into a structure which is of fixed size and shape in the assembly loader. For instance, we effectively have that loop now in R2R, and I've never seen it have a high cost that matters anywhere.

Reply:
OK, that sounds reasonable, even Bing with its 1K-ish number of assemblies shouldn't pose a big problem for parsing a simple runtime structure and filling in some internal array of structures in the CoreCLR runtime.

Reply:
In 2nd commit I have changed the proposed design so that the R2R header is basically two-level: for composite images, the master header includes the section <code>READYTORUN_SECTION_ASSEMBLIES</code> that for each component assembly points at its own sub-header with a list of sections.

Reply:
@trylek If that number were 10K would it still be reasonable to have it this way? I'm not suggesting 10K be a benchmark but it would be nice to either know this is a one-time startup cost and therefore it is ok to pay that for simplicity.

Reply:
@mjsabby My personal feeling is that, even with 10K assemblies, other costs still far outweigh parsing the headers and storing the information somewhere; in general you're right however, the easiest way to improve asymptotic startup is by minimizing any forms of runtime preprocessing.